virtcontainers/sandbox.go

```diff
@@ -939,6 +939,12 @@ func (s *Sandbox) startVM() error {
 		return err
 	}
 
+	defer func() {
+		if err != nil {
+			s.hypervisor.stopSandbox()
+		}
+	}()
```

Comment:
Is this new `defer` function fixing the issue you're seeing? The patch looks good here, and if it's fixing the problem, then let's merge it.

Reply:
> run many container parallel may reproduce, but it also simple to reproduce if you created an error in startVM(), the qemu will leak

@sboeuf ， like I said in the https://github.com/kata-containers/runtime/issues/1426, if you donnot mind, you can create an error in the startVM function, and should see the fail

Reply:
You didn't answer my question here. Is the current version of this PR fixing the issue that you are seeing when running many containers in parallel?
If it is, then please let me know and I'll merge the patch.
If it's not, then extend this patch until you get it fixed.

Reply:
Yes, it does fix the problem here, and I provide a way to reproduce the problem.

Reply:
Great, then let's merge it.